mlrun/api/launcher.py

```diff
@@ -194,6 +204,22 @@ def enrich_runtime(
                 project, runtime, copy_function=False
             )
 
+    def _enrich_before_run(
```

Comment:
```suggestion
    def _enrich_full_spec(
```
Or maybe a more indicative name of what the method is doing, and not just when it's run. As maybe we will want more enrichments that don't need the full param.